src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -463,6 +493,35 @@ private string ConfigureDatabase(InitializeViewRequestParams viewParams, Databas
                                 db110.DatabaseContainmentType = containmentTypeEnums[database.ContainmentType];
                             }
                         }
+                        if (prototype is DatabasePrototype130 db130)
+                        {
+                            if (databaseScopedConfigurationsCollection != null)
+                            {
+                                foreach (DatabaseScopedConfigurationsInfo dsc in database.DatabaseScopedConfigurations)
+                                {
+                                    foreach (DatabaseScopedConfiguration smoDscCollection in databaseScopedConfigurationsCollection)
+                                    {
+                                        if (smoDscCollection.Name == dsc.Name)
+                                        {
+                                            smoDscCollection.Value = dsc.ValueForPrimary == SR.prototype_db_prop_databasescopedconfig_value_fail_enabled
+                                                ? "1" : dsc.ValueForPrimary == SR.prototype_db_prop_databasescopedconfig_value_fail_disabled
+                                                ? "0" : dsc.ValueForPrimary;
+
+                                            // seconday value needs to set back to 'PRIMARY' to avoid unnecessary script generation for few properties
```

Comment:
"for few properties" - what does this mean? If any of the values are changed shouldn't we be generating the script?

Reply:
I'll update the comment more meaningful!

FYI- We display primaryValue to the secondary when it has the 'PRIMARY' value. When we are sending back to SMO, we must update back to PRIMARY if not changed, so that smo would not generate alter dsc for secondary script, otherwise it generates the below unnecessary syntax. 

ALTER DATABASE SCOPED CONFIGURATION FOR SECONDARY SET **DSC_PROPERTY_HERE** = OFF